plugins/woocommerce/includes/admin/class-wc-admin-log-table-list.php

```diff
@@ -255,6 +276,83 @@ protected function source_dropdown() {
 		}
 	}
 
+	/**
+	 * Generates the table rows.
+	 *
+	 * @since 3.1.0
+	 */
+	public function display_rows() {
+		foreach ( $this->items as $item ) {
+			$this->single_row( $item );
+			$this->single_row_log( $item );
+		}
+	}
+
+	/**
+	 * Generates content for a single row of the table.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @param object|array $item The current item.
+	 */
+	public function single_row_log( $item ) {
+		// Maintains alternating row background colors.
+		?>
+		<tr style="display: none"><td></td></tr>
+		<tr class="details-box" data-details-box="<?php echo esc_attr( $item['log_id'] ); ?>" style="display: none">
+			<td colspan="<?php echo intval( $this->get_column_count() ); ?>">
+				<div style="display: flex; flex-flow: row wrap; column-gap: 20px;">
+					<div style="flex-grow:1">
+						<p style="font-weight: bold"><?php echo esc_html_e( 'Order Details', 'woocommerce' ); ?></p>
+						<?php $this->render_log( 'order', $item ); ?>
+					</div>
+					<div style="flex-grow:1">
+						<p style="font-weight: bold"><?php echo esc_html_e( 'Error Details', 'woocommerce' ); ?></p>
+						<?php $this->render_log( 'error', $item ); ?>
+					</div>
+				</div>
+			</td>
+		</tr>
+		<?php
+	}
+
+	/**
+	 * Render the Log Deails.
+	 *
+	 * @param string $key The key name.
+	 * @param array  $log The Log item.
+	 */
+	public function render_log( $key, $log ) {
+		// This is unserializing "safe" data from the database, not user input.
+		$context = unserialize( $log['context'] );
```

Comment:
👋🏽 Apologies for the delay. We haven't forgotten about this PR. I'd like to pause and get some further opinions here, as this is a *potential* safety issue (despite the fact it is coming from the database).

I was curious about the using `unserialize()`'s optional `allowed_classes` option as a way of adding safety (we could allow only `stdClass`) but in testing this out it is apparent that that will drop some amount of information, which would defeat the purpose of this change.

A better option may be to switch to using JSON encoding going forward, since this would alleviate safety concerns and it would also be easy to differentiate between JSON and serialized PHP at this stage (we'd avoid visualizing serialized PHP). However, give me a chance to seek some further opinions here before we do more on this :+1:

Reply:
@johnregan3: sorry for the delay.

After talking this out, how about we simply present the context in its serialized form? It will still surface key information (which could for example still be shared with plugin support), is safe, and—if we wished—we might even find a way to pretty-print it.

What are your thoughts?

Reply:
@barryhughes We can totally do that. I'd like do this as safely as possible. I'll look into pretty-printing this out.

Reply:
Excellent, thank you @johnregan3.

Reply:
As soon as you're ready, please simply re-request a review :+1: